### ISSUE TITLE
[Bug]: Screenshot masking not working for fullpage

### ISSUE THREAD

**Issue by sloeff** [microsoft/playwright#31060]
### Version

1.44

### Steps to reproduce

I cannot share the SUT since its running on a .local domain but I do my best to share enough info to reproduce the issue. Just DONT outright close it when information is lacking. Just ask for info that you require.
The problem started when updating to Playwright v 1.44, 1.42.1 worked fine. We do full page visual comparsions since we have a lot of functionality for the responsive site. Since 1.44 we see that not all elements we want to mask are being masked. At some point Playwright stops masking which obviously creates failing tests. (we mask elements for a reason).

Expected screenshot was made in Playwright v1.42.1, actual is Playwright v1.44. (Please ignore the black markers)
![Comparison](https://github.com/microsoft/playwright/assets/18349193/dc203d27-04e0-4c0e-8287-1d2fa5614a64)

Repo: https://github.com/sloeff/playwright_issue

Reproduction steps:

1. I recommend using a page with the same element repeating lots of times over the hole page.
2. Change the mask Locator to this element
3. Run the test and check the test report
4. Notice in the report that not all element that are marked for masking are masked

### Expected behavior

Masking is correctly over the specified elements on the full page

### Actual behavior

The mask option within toHaveScreenshot doesn't mask all elements on the page. The top elements get masked futher down the page the elements are not longer masked.

### Environment

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (4) x64 Intel(R) Xeon(R) Platinum 8276 CPU @ 2.20GHz
    Memory: 8.46 GB / 16.00 GB
  Binaries:
    Node: 16.20.1 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.4 - C:\Program Files\nodejs\npm.CMD
  Managers:
    Maven: 3.8.2 - D:\tools\apache-maven-3.8.2\bin\mvn.CMD
  Utilities:
    Git: 2.39.2.
    Curl: 8.4.0 - C:\WINDOWS\system32\curl.EXE
  IDEs:
    VSCode: 1.89.1 - C:\Users\<redacted>\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
  Languages:
    Bash: 5.2.12 - C:\Program Files\Git\bin\bash.EXE
    Java: 1.8.0_341
  Browsers:
    Edge: Chromium (125.0.2535.51)
    Internet Explorer: 11.0.19041.4355
```

**Comment by dgozman** [comment]
This looks like a duplicate of #30770, which was already fixed in v1.44.1. Please update Playwright to the latest version and let me know whether that helped.

**Comment by sloeff** [comment]
Yes, the issue is resolved. Thanks!
